components/DetailsCard/index.js

```diff
@@ -27,8 +27,16 @@ export default function DetailsCard() {
 
   async function handleEditActivity(event) {
     event.preventDefault();
-    const formData = new FormData(event.target);
-    const activityData = Object.fromEntries(formData);
+    // const formData = new FormData(event.target);
+    // const activityData = Object.fromEntries(formData);
+
+    const form = event.target;
+    const formData = new FormData(form);
+
+    const activityData = {};
+    formData.forEach((value, key) => {
+      activityData[key] = value;
+    });
```

Comment:
This should lead to the same result, while using less code: 

```suggestion
const activityData = Object.fromEntries(formData.entries());
```